ts/components/session/SessionIconButton.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import classNames from 'classnames';
+
+export enum SessionIconButtonTypes {
+  'exit' = 'exit',
+  'search' = 'search',
+  'back' = 'back',
+  'attachment' = 'attachment',
+  'emoji' = 'emoji',
+  'favorite' = 'favorite',
+  'group' = 'group',
+  'menu' = 'menu',
+  'message' = 'message',
+  'microphone' = 'microphone',
+  'network' = 'network',
+  'options' = 'options',
+  'theme' = 'theme',
+}
+
+export enum SessionIconButtonSizes {
+  'small' = 'small',
+  'medium' = 'medium',
+  'large' = 'large',
+}
+
+interface Props {
+  iconType: SessionIconButtonTypes;
+  iconSize: SessionIconButtonSizes;
+}
+
+export class SessionIconButton extends React.PureComponent<Props> {
+  constructor(props: any) {
+    super(props);
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  public render() {
+    const { iconType, iconSize } = this.props;
+
+    const iconPath = `./images/session/icon-${iconType}.svg`;
+
+    return (
+      <div
+        className={classNames(
+          'session-icon-button',
+          iconType === SessionIconButtonTypes.exit ? 'exit' : '',
+          iconType === SessionIconButtonTypes.search ? 'search' : '',
+          iconType === SessionIconButtonTypes.back ? 'back' : '',
+          iconSize === SessionIconButtonSizes.small ? 'small' : '',
+          iconSize === SessionIconButtonSizes.medium ? 'medium' : '',
+          iconSize === SessionIconButtonSizes.large ? 'large' : ''
+        )}
+        role="button"
+        onClick={this.clickHandler}
+      >
+        <img src={iconPath} alt="" />
+      </div>
+    );
+  }
+
+  private clickHandler() {
```

Comment:
this clickhandler needs to forward the event to whatever the parent set in the props.

(I just did the same for the SessionButton)


main idea is:  in your `session-icon-button` `div` put:

```
onClick={e => {
          this.clickHandler(e);
        }}
```

in the handler:

```  
private clickHandler(e: any) {
    if (this.props.onClick) {
      e.stopPropagation();
      this.props.onClick()
    }
  }
```

and add it to the prop:

`  onClick: any;`

Reply:
Nice mate. Will update